cpu_scheduling_algorithms/rr_scheduling.cpp

```diff
@@ -0,0 +1,89 @@
+
+// Round Robin is a scheduling algorithm.
+// In Round Robin each process is assigned a fixed time slot in a cyclic way.
+// https://en.wikipedia.org/wiki/Round-robin_scheduling
+#include <iostream>
+#include <vector>
+#include <numeric>
+#include <string>
+#include <functional>
+
+/*----------------------------------------------------------------
+// Calculate the waiting times of a list of processes that have a
+//        specified duration.
+//
+//        Return: The waiting time for each process.
+//        >>> calculate_waiting_times([10, 5, 8])
+//        [13, 10, 13]
+//        >>> calculate_waiting_times([4, 6, 3, 1])
+//        [5, 8, 9, 6]
+//        >>> calculate_waiting_times([12, 2, 10])
+//        [12, 2, 12]
```

Comment:
For documenting the parameters, return statements, and detailed description of the function, the following format should be used:

```c++
...
 * @tparam T this is a one-line info about T
 * @param param1 on-line info about param1
 * @param param2 on-line info about param2
 * @returns `true` if ...
 * @returns `false` if ...
 */
```